### PR TITLE
[PM-9409] Add FIDO 2 authentication to credential manager

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

PM-9409

## 📔 Objective

Implement FIDO 2 credential assertion using the FIDO 2 credential manager.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by dseverns-livefront — APPROVED** [review]

File: app/src/test/java/com/x8bit/bitwarden/data/vault/datasource/sdk/util/PublicKeyCredentialAuthenticatorAssertionResponseExtensionsTest.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                rk = null,
+                authenticatorDisplayName = null,
+            ),
+        )
+        val result = mockSdkResponse.toAndroidFido2PublicKeyCredential()
+        assertTrue(result.clientExtensionResults.credentialProperties?.residentKey!!)
+    }
+}
+
+private fun createMockSdkAssertionResponse(
```
Comment (dseverns-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/vault/datasource/sdk/util/PublicKeyCredentialAuthenticatorAssertionResponseExtensionsTest.kt:85]: Nice call!

**Review by dseverns-livefront — APPROVED** [review]

**Review by dseverns-livefront — APPROVED** [review]

File: app/src/main/java/com/x8bit/bitwarden/ui/autofill/fido2/manager/Fido2CompletionManagerImpl.kt [hunk 2]
```diff
@@ -63,14 +62,6 @@ class Fido2CompletionManagerImpl(
         activity.also {
             val intent = Intent()
             when (result) {
-                Fido2CredentialAssertionResult.Cancelled -> {
```
Comment (shannon-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/autofill/fido2/manager/Fido2CompletionManagerImpl.kt:66]: What's the reason for deleting this case?

Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/ui/autofill/fido2/manager/Fido2CompletionManagerImpl.kt:66]: Great question. There is no opportunity for the assertion to be explicitly cancelled, short of backing out of the app, which responds to the OS with a cancellation signal. The user will only see Bitwarden while verification takes place. Cancelling or failing verification will display an error dialog to the user and complete the process by reporting an error to the OS. See the clip from #3630 below for an example.

passkey-assertion-unlocked

Comment (shannon-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/autofill/fido2/manager/Fido2CompletionManagerImpl.kt:66]: Very helpful, thank you!

**Pull request merged**